tests/test_enterprise/api/test_views.py

```diff
@@ -1679,26 +1679,43 @@ class TestEnterpriseCustomerCatalogWriteViewSet(BaseTestEnterpriseAPIViews):
     Test EnterpriseCustomerCatalogWriteViewSet
     """
     @ddt.data(
-        (False, 403),
-        (True, 201),
+        (False, 403, False),
+        (True, 201, False),
+        (True, 200, True),
     )
     @ddt.unpack
-    def test_create_catalog(self, is_staff, expected_status_code):
+    def test_create_catalog(self, is_staff, expected_status_code, isIdempotent):
         """
         Test that a catalog can be created
         """
         enterprise_customer = factories.EnterpriseCustomerFactory(uuid=FAKE_UUIDS[0])
+        enterprise_catalog_query = factories.EnterpriseCatalogQueryFactory(uuid=FAKE_UUIDS[0])
 
         self.set_jwt_cookie(ENTERPRISE_ADMIN_ROLE, str(enterprise_customer.uuid))
         self.user.is_staff = is_staff
         self.user.save()
+
         response = self.client.post(ENTERPRISE_CUSTOMER_CATALOG_ENDPOINT, {
             "title": "Test Catalog",
             "enterprise_customer": str(enterprise_customer.uuid),
+            "enterprise_catalog_query": str(enterprise_catalog_query.id),
         }, format='json')
         response_output = self.load_json(response.content)
 
-        assert response.status_code == expected_status_code
+        if isIdempotent:
```

Comment:
I like that you attempted to reuse the DDT stuff here but I'm not sure that this is a good use case. I find this test hard to read and it's not immediately clear what is being tested and when. I would suggest writing a new test the deals only with the indempotency bits and to leave this test alone. 

Also I think that testing for idempotency warrants it's own test.